numpy/_core/tests/test_custom_dtypes.py

```diff
@@ -231,6 +231,30 @@ def test_wrapped_and_wrapped_reductions(self):
         expected = np.hypot.reduce(float_equiv, keepdims=True)
         assert res.view(np.float64) * 2 == expected
 
+    def test_sort(self):
+        a = self._get_array(1.)
+        a = a[::-1]  # reverse it
+
+        a.sort()
+        assert_array_equal(a.view(np.float64), [1., 2., 3.])
+
+        a = self._get_array(1.)
+        a = a[::-1]  # reverse it
+
+        sorted_a = np.sort(a)
+        assert_array_equal(sorted_a.view(np.float64), [1., 2., 3.])
+        # original is unchanged
+        assert_array_equal(a.view(np.float64), [3., 2., 1.])
+
+    def test_argsort(self):
+        a = self._get_array(1.)
+        a = a[::-1]  # reverse it
+
+        indices = np.argsort(a)
+        assert_array_equal(indices, [2, 1, 0])
+        # original is unchanged
+        assert_array_equal(a.view(np.float64), [3., 2., 1.])
```

Comment:
Would be good to try to do some tests with swapped byte order, weird strides, etc.

Reply:
Yes, added a scaled float with a stride of 2! Unfortunately it seems a bit hard to get byte-order flipped sfloats - there doesn't seem to be a dtype flip implemented. I can implement it if the test would be really good to have.